perses/tests/test_topology_proposal.py

```diff
@@ -905,20 +905,144 @@ def test_simple_heterocycle_mapping(iupac_pairs = [('benzene', 'pyridine')]):
 
         assert num_hetero_maps > 0, f"there are no differences in atomic number mappings in {iupac_pair}"
 
+def test_protein_counterion_topology_fix_positive():
```

Comment:
@zhang-ivy can you check this test and the other two that I added? The bits I changed:

* changed `charge_diffs` to `charge_diff_test` 
* removed `counterion_name` since it was unused
*  added a negative and zero test for charges

Reply:
Looks good to me!